devito/types/basic.py

```diff
@@ -611,20 +599,20 @@ class AbstractFunction(sympy.Function, Basic, Cached, Pickable, Evaluable):
     is_imaginary = False
     is_commutative = True
 
-    @classmethod
-    def _cache_key(cls, *args, **kwargs):
-        """An AbstractFunction caches on the class type itself."""
-        return cls
-
     def __new__(cls, *args, **kwargs):
         options = kwargs.get('options', {'evaluate': False})
 
-        key = cls._cache_key(*args, **kwargs)
-        obj = cls._cache_get(key)
+        # Is it an existing shifted index?
+        obj = cls._cache_get((cls, *args))
```

Comment:
so here `args==indices` basically?

Reply:
Yes, at subs/new/.... it's called as `func(*indices, **kwargs)`